presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcIdentity.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.jdbc;
+
+import com.google.common.collect.ImmutableMap;
+import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.security.ConnectorIdentity;
+
+import java.util.Map;
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class JdbcIdentity
+{
+    public static JdbcIdentity from(ConnectorSession session)
+    {
+        return from(session.getIdentity());
+    }
+
+    public static JdbcIdentity from(ConnectorIdentity connectorIdentity)
+    {
+        return new JdbcIdentity(connectorIdentity.getExtraCredentials());
+    }
+
+    private final Map<String, String> extraCredentials;
+
+    public JdbcIdentity(Map<String, String> extraCredentials)
+    {
+        this.extraCredentials = ImmutableMap.copyOf(requireNonNull(extraCredentials, "extraCredentials is null"));
+    }
+
+    public Map<String, String> getExtraCredentials()
+    {
+        return extraCredentials;
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        JdbcIdentity that = (JdbcIdentity) o;
+        return Objects.equals(extraCredentials, that.extraCredentials);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(extraCredentials);
+    }
+
+    @Override
+    public String toString()
+    {
+        return toStringHelper(this)
+                .add("extraCredentials", extraCredentials)
```

Comment:
I think we only want to include the keys here, like in `Identity`, as the values could be passwords or other sensitive things.